lib/_http_server.js

```diff
@@ -213,8 +213,25 @@ ServerResponse.prototype.writeHeader = function() {
 };
 
 
-function Server(requestListener) {
-  if (!(this instanceof Server)) return new Server(requestListener);
+var TLS_ARGS = ['pfx', 'key', 'passphrase', 'cert', 'ca', 'crl', 'ciphers',
```

Comment:
If we're planning on adding an `options` for `http.createServer()` in the future, then if nothing else we could just check for one of the required https config options (e.g. `pfx` and `key`) to determine if an `https.Server` should be created.